cms/envs/aws.py

```diff
@@ -534,3 +534,15 @@
 
 # Allow extra middleware classes to be added to the app through configuration.
 MIDDLEWARE_CLASSES.extend(ENV_TOKENS.get('EXTRA_MIDDLEWARE_CLASSES', []))
+
+########################## Comprehensive Theming  #######################
+
+# Set up comprehensive theming after all other settings have been set to avoid
+# modifying paths before ENABLE_COMPREHENSIVE_THEMING has its final value.
+if ENABLE_COMPREHENSIVE_THEMING:
+    LOCALE_PATHS = enable_theming_with_settings(
+        MAKO_TEMPLATES['main'],
+        get_theme_base_dirs_from_settings(COMPREHENSIVE_THEME_DIRS),
+        LOCALE_PATHS,
+        COMPREHENSIVE_THEME_LOCALE_PATHS
+    )
```

Comment:
This code is what I've added to the ends of all relevant settings files. The section needs to execute after the final value of `ENABLE_COMPREHENSIVE_THEMING` has been set. It's ugly and I'm open to suggested improvements!

Reply:
In a settings refactor on a previous project, I added a `derived` settings module which I recall featuring two main functions:

* A function for adding the name and calculation function/lambda for a setting which depended on other settings
* A function to calculate all the derived settings at the end of each top-level settings file, passing the settings so far in as an argument.

This kept the usual style of being able to group related settings together, but some of them only got a final value after the ones they depended on had been through all possible override points.  It worked fairly well, but I'm not sure if we want to adopt a custom solution like that here vs. looking for an off-the-shelf solution.  I don't know off hand if libraries like [django-configurations](https://github.com/jazzband/django-configurations) or [grift](https://github.com/kensho-technologies/grift) have a good solution for this.